core/renderers/zelos/info.js

```diff
@@ -239,31 +239,42 @@ Blockly.zelos.RenderInfo.prototype.getElemCenterline_ = function(row, elem) {
  * @protected
  */
 Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
-  if (!this.topRow.hasPreviousConnection) {
-    return;
-  }
-  var minXPos = this.constants_.NOTCH_OFFSET_LEFT +
+  var notchTotalWidth = this.constants_.NOTCH_OFFSET_LEFT +
       this.constants_.NOTCH_WIDTH;
-  for (var i = 0, row; (row = this.rows[i]); i++) {
-    if (Blockly.blockRendering.Types.isInputRow(row)) {
+  var minXPos = notchTotalWidth;
+  for (var i = 2; i < this.rows.length - 1; i += 2) {
```

Comment:
Might help readability to have a comment explaining the goal of this for loop. Specifically, might be nice to know why we are counting by 2s.